Description: use long literal for CURLOPT_POSTFIELDSIZE
Author: Ruohang Feng <rh@vonng.com>
Forwarded: not-yet
Last-Update: 2026-02-20

--- a/src/backend/access/external/url_curl.c	2025-08-06 12:12:30
+++ b/src/backend/access/external/url_curl.c	2026-02-20 15:47:23
@@ -1473,7 +1473,7 @@
 	{
 		/* use empty message */
 		CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDS, "");
-		CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDSIZE, 0);
+		CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDSIZE, 0L);
 
 		/* post away and check response, retry if failed (timeout or * connect error) */
 		gp_perform_backoff_and_check_response(file, easy_perform_work);
@@ -1932,7 +1932,7 @@
 
 	/* use empty message */
 	CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDS, "");
-	CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDSIZE, 0);
+	CURL_EASY_SETOPT(file->curl->handle, CURLOPT_POSTFIELDSIZE, 0L);
 
 	/* post away! */
 	gp_perform_backoff_and_check_response(file, easy_perform_work);
