diff --git a/pg_dbms_lock.control b/pg_dbms_lock.control
index 460ddf4..d631319 100644
--- a/pg_dbms_lock.control
+++ b/pg_dbms_lock.control
@@ -1,5 +1,6 @@
 comment = 'Extension to add Oracle DBMS_LOCK full compatibility to PostgreSQL'
 default_version = '2.0.0'
 module_pathname = '$libdir/pg_dbms_lock'
+requires = 'pg_background'
 schema = 'dbms_lock'
 relocatable = false
diff --git a/sql/pg_dbms_lock--2.0.0.sql b/sql/pg_dbms_lock--2.0.0.sql
index dd0f020..b051e8d 100644
--- a/sql/pg_dbms_lock--2.0.0.sql
+++ b/sql/pg_dbms_lock--2.0.0.sql
@@ -66,9 +66,9 @@ BEGIN
         -- Create an entry in the dbms_lock.dbms_lock_allocated table
         -- that will be completed by the call to dbms_lock.request
         query := format('INSERT INTO dbms_lock.dbms_lock_allocated VALUES (%L, %s, %L) ON CONFLICT (name) DO UPDATE SET expiration = %L', lockname, lockhandle_id, expire_date::timestamp(0), expire_date::timestamp(0));
-        query := 'SELECT (h).pid, (h).cookie FROM pg_background_launch(' || quote_literal(query) || ') as h';
+        query := 'SELECT (h).pid, (h).cookie FROM pg_background_launch_v2(' || quote_literal(query) || ') as h';
         EXECUTE query INTO lockpid, lockcookie_id;
-        query := 'SELECT * FROM pg_background_result(' || lockpid || ', ' || lockcookie_id || ') as (result TEXT)';
+        query := 'SELECT * FROM pg_background_result_v2(' || lockpid || ', ' || lockcookie_id || ') as (result TEXT)';
         EXECUTE query;
 
 	-- Set output parameter
diff --git a/t/03_sessions.t b/t/03_sessions.t
index ab70c87..8e73669 100755
--- a/t/03_sessions.t
+++ b/t/03_sessions.t
@@ -15,7 +15,7 @@ ok( $? == 0, "test for session2 lock");
 $ret = `psql -X -d regress_dbms_lock -f test/sql/session3.sql > results/session3.out 2>&1`;
 ok( $? == 0, "test for session3 lock");
 
-sleep(7);
+sleep(15);
 
 $ret = `psql -X -d regress_dbms_lock -c "SELECT action, session, (date_trunc('second', executed) - lag(date_trunc('second', executed)) over(PARTITION BY session)) FROM lock_test" > results/session.out`;
 ok( $? == 0, "result for sessions 2 and 3 work");
diff --git a/updates/pg_dbms_lock--1.0.0--2.0.0.sql b/updates/pg_dbms_lock--1.0.0--2.0.0.sql
index f0f683f..8965376 100644
--- a/updates/pg_dbms_lock--1.0.0--2.0.0.sql
+++ b/updates/pg_dbms_lock--1.0.0--2.0.0.sql
@@ -35,9 +35,9 @@ BEGIN
         -- Create an entry in the dbms_lock.dbms_lock_allocated table
         -- that will be completed by the call to dbms_lock.request
         query := format('INSERT INTO dbms_lock.dbms_lock_allocated VALUES (%L, %s, %L) ON CONFLICT (name) DO UPDATE SET expiration = %L', lockname, lockhandle_id, expire_date::timestamp(0), expire_date::timestamp(0));
-        query := 'SELECT (h).pid, (h).cookie FROM pg_background_launch(' || quote_literal(query) || ') as h';
+        query := 'SELECT (h).pid, (h).cookie FROM pg_background_launch_v2(' || quote_literal(query) || ') as h';
         EXECUTE query INTO lockpid, lockcookie_id;
-        query := 'SELECT * FROM pg_background_result(' || lockpid || ', ' || lockcookie_id || ') as (result TEXT)';
+        query := 'SELECT * FROM pg_background_result_v2(' || lockpid || ', ' || lockcookie_id || ') as (result TEXT)';
         EXECUTE query;
 
 	-- Set output parameter
